cfpb/cfgov-refresh
# Ask CFPB `read more` functionality

Extends expandable to allow `read more` preview of answer content on mobile screens

## Additions

- Expandable modifier in  `read-more.less` 
- Read more expandable html in `answer-page.html`
- Set up/tear down of read more expandable at mobile sizes in `ask-cfpb/single.js`

## Changes

- Make resize handler conditional in `Expandable.js` so height is not set for read more content on larger screen sizes

## Testing

- Check out branch
- Set up Ask if necessary
  ```
  export DEPLOY_ENVIRONMENT=build
  ./drop-db.sh
  ./create-mysql-db.sh
  mysql -uroot v1 < ~/Downloads/production_django.sql
  python cfgov/manage.py migrate
  python cfgov/manage.py runscript migrate_knowledgebase
  ```
- Navigate to [Ask answer page](http://localhost:8000/ask-cfpb/what-effect-will-shopping-for-an-auto-loan-have-on-my-credit-en-763/)
  - Verify that read more expandable is activated on mobile screen sizes and deactivated when expandable is expanded
  - Verify that read more expandable is not activated on larger screens


## Screenshots
### Read more on mobile
<img width="380" alt="screen shot 2017-05-30 at 8 00 37 pm" src="https://cloud.githubusercontent.com/assets/778171/26614334/83c2288a-4574-11e7-8756-15b2b941cdc9.png">

### Larger screen sizes
<img width="1132" alt="screen shot 2017-05-30 at 8 00 48 pm" src="https://cloud.githubusercontent.com/assets/778171/26614335/864adbba-4574-11e7-95d0-6d783ca601aa.png">


## Checklist
